src_c/math.c

```diff
@@ -800,6 +804,69 @@ vector_copy(pgVector *self)
     return (PyObject *)ret;
 }
 
+static PyObject *
+vector_clamp_magnitude(pgVector *self, PyObject *args, PyObject *kwargs)
+{
+    Py_ssize_t i;
+    pgVector *ret;
+
+    ret = (pgVector *)pgVector_NEW(self->dim);
+    if (ret == NULL)
+        return NULL;
+
+    for (i = 0; i < self->dim; ++i)
+        ret->coords[i] = self->coords[i];
+
+    vector_clamp_magnitude_ip(ret, args, kwargs);
```

Comment:
```suggestion
    PyObject *ret_val = vector_clamp_magnitude_ip(ret, args, kwargs);
    if (!ret_val) {
        return NULL;
    }
    Py_DECREF(ret_val);
```
There's a memory leak and an error checking issue here. When `vector_clamp_magnitude_ip` fails it returns `NULL`, so this function should return `NULL` too when that happens. When `vector_clamp_magnitude_ip` returns python object `None` after a successful call, it should be `Py_DECREF`d like any other python object